tools/deployment_options.py

```diff
@@ -30,6 +30,12 @@ def load_deployment_options(parser=None):
         choices=['minikube', 'oc', 'oc-ingress', 'ocp'],
         default='minikube'
     )
+    parser.add_argument(
+        '--persistent-storage',
+        help='Persistent storage is used',
+        type=bool,
+        default=True
+    )
```

Comment:
```suggestion
    parser.add_argument(
        '--persistent-storage',
        help='Persistent storage is used',
        action=store_true
    )
```

I may have misunderstood what you are after here but, from what I can tell I think what you want is to use `--persistent-storage` and `--apply-manifest` as flags. In this case, using `store_true` as an action is better. Using `type` is normally not recommended, especially with the `bool` converter as it will convert empty strings to False and whatever value to True.

Reply:
They aren't meant to be flags.  You did point out a bug in how --persistent-storage was being used. The Makefile wasn't passing in a value like --persistent-storage $(PERSISTENT_STORAGE) in the Makefile to deploy_postgres.py, and that was probably the source of the confusion.

Good to know about action='store_true'. For me, It was counter-intuitive that the default value would be False.

Reply:
> They aren't meant to be flags.

I'm a bit confused by this because they are currently being used as flags. The store true/false and they are used as booleans everywhere else. Maybe I'm missing something.

Could you elaborate on what cases these CLI params won't be used as flags? I'm sure there's a learning opportunity for me here :smile: